### DISCUSSION TITLE
Setup, First Steps

### DISCUSSION THREAD

**Discussion by pwoolvett** [flakeheaven/flakeheaven#1] [Governance]
Hi all!

I'm closing [pw/flakehell](https://github.com/pwoolvett/flakehell/) in favor of this, so I think this is the best place to continue with [the old tasks](https://github.com/pwoolvett/flakehell/projects/1)

## 1. The fork

I've talked with the original creator, and he's fine with us changing name, logos, references, etc -  as long as we 
1.  [x] add a reference on `orsinium/forks`(https://github.com/orsinium/forks)
2. [x] add a reference on the readme

## 2. Documentation

1. [x] Documentation should be hosted at readthedocs/netlify/equivalent, and latest stable (master, see below) hosted in gh-pages as well
       * [x] readthedocs integration enabled - served at [read the docs](https://flakeheaven.readthedocs.io/)
2. [x] Connect static docs service to PR so they could be previewed before publishing
       * [x] readthedocs pull requests integration enabled
3. alternatively, keep all versions of the documentation in gh-pages
4. newsfragment + automatic changelog (https://pypi.org/project/towncrier/, see pypa/pip, https://github.com/pytest-dev/pytest/blob/master/pyproject.toml)
5. Badges
  - [x] fury -> pypi
  - [x] drone -> github actions
  - [x] readthedocs -> readthedocs?

## 3. Code style

1. I'd like to move code inside a `src` directory. It avoids some problems for newcomers (eg `import flakeheaven` when its in `PYTHONPATH`)
2. Move `docs` to `docs/src`
3. replace dephell (also archived) with poetry
4. replace drone ci with github actions 
   1. [ ] format
   2. [x] lint
   3. [x] test( pytest+pytest-cov)
   4. docs (sphinx, sphinx-apidoc, myst, ...)
5. enforce conventional commits and autoversioning
6. disable master push

## 4. Forks

Maybe we could import PRs from other repos (eg  [flakehell/flakehell)](https://github.com/flakehell/flakehell)) )

## 5. Kanban

After some discussion, we should move this to a project to start working on stuff.

## 6. Others

* [x] I still own flakeheaven on pypi, so when we're ready to publish i'll release the name so it can be used for this project.
* how should we maintain pypi? I think a single user with token saved as github secret should be enough

**Comment by mcarans** [comment]
I think that all you've proposed would be a big improvement over the current setup. I would like to propose one change though and that is getting out a version of flakeheaven with my PR for Flake8 4.0.x from here: https://github.com/flakehell/flakehell/pull/23 as soon as is reasonably possible so that people will start using this project rather than giving up and moving on to other solutions. 

I worry that given you've said you have limited time and I also am in that situation that if we try to implement all of 1, 2 and 3 before making a release that works with Flake8 4.0.x, that people will have given up on flakehell/heaven as a solution and found alternatives before too long. Please let me know your thoughts on this @pwoolvett and if needed, I can create the PR.

For docs, I quite like pydoc-markdown + readthedocs, but the solutions you've identified sound good to me as well. I'm not familiar with poetry, having used a pyproject.toml and setup.cfg solution myself but again a modern setup like you've described would be great (example setup for one of my projects with pyproject, tox etc.: https://github.com/OCHA-DAP/hdx-python-api)

**Reply by jose-rojas0** [reply]
I agree here - we should try to get an MVP out with support for flake8 4.0.x so that people using `flakehell` land here. After that I will likely be able to help with a few of the items listed here (poetry and github actions in particular I am fairly comfortable working with)

**Reply by lyz-code** [reply]
Yep, I agree too, I can spend some time on flakeheaven, but it's not going to be my first priority, so I'd just invest time on maintenance stuff instead of new features

**Comment by pwoolvett** [comment]
ok so @mcarans create the pr

as the minimum, i'd aim for:

* Tests for flake8 >= 4 (is the patch retrocompatible?)
* Change flakehell -> flakeheaven
* readme (archive, ref upstream, wtc)

(also add this project to orsiniums forks as required, not related to the pr) 

Once we got that covered, well merge it to master, tag 1.0.0 and upload manually to pypi using twine

**Reply by mcarans** [reply]
I have made the PR: https://github.com/flakeheaven/flakeheaven/pull/3  Please take a look

**Comment by pwoolvett** [comment]
* flakeheaven is now owned by flakeheaven, not me - both  on testpypi, and pypi
* first version is `0.11.0`, with @mcarans support for flake8>4
* https://discord.gg/6zKFFKMf

**Reply by pwoolvett** [reply]
flakeheaven connected to [readthedocs](https://flakeheaven.readthedocs.io/)
